src/components/AvatarWithImagePicker.js

```diff
@@ -116,166 +116,118 @@ const defaultProps = {
+    };
+
+    const isValidResolution = async (image) => {
+        const resolution = await getImageResolution(image);
+        return resolution.height >= CONST.AVATAR_MIN_HEIGHT_PX &&
+            resolution.width >= CONST.AVATAR_MIN_WIDTH_PX &&
+            resolution.height <= CONST.AVATAR_MAX_HEIGHT_PX &&
+            resolution.width <= CONST.AVATAR_MAX_WIDTH_PX;
+    };
+
+    const showAvatarCropModal = async (image) => {
+        if (!isValidExtension(image)) {
+            setError('avatarWithImagePicker.notAllowedExtension', { allowedExtensions: CONST.AVATAR_ALLOWED_EXTENSIONS });
             return;
         }
-        if (!this.isValidSize(image)) {
-            this.setError('avatarWithImagePicker.sizeExceeded', {maxUploadSizeInMB: CONST.AVATAR_MAX_ATTACHMENT_SIZE / (1024 * 1024)});
+        if (!isValidSize(image)) {
+            setError('avatarWithImagePicker.sizeExceeded', { maxUploadSizeInMB: CONST.AVATAR_MAX_ATTACHMENT_SIZE / (1024 * 1024) });
             return;
         }
 
-        this.isValidResolution(image).then((isValidResolution) => {
-            if (!isValidResolution) {
-                this.setError('avatarWithImagePicker.resolutionConstraints', {
-                    minHeightInPx: CONST.AVATAR_MIN_HEIGHT_PX,
-                    minWidthInPx: CONST.AVATAR_MIN_WIDTH_PX,
-                    maxHeightInPx: CONST.AVATAR_MAX_HEIGHT_PX,
-                    maxWidthInPx: CONST.AVATAR_MAX_WIDTH_PX,
-                });
-                return;
-            }
-
-            this.setState({
-                isAvatarCropModalOpen: true,
-                validationError: null,
-                phraseParam: {},
-                isMenuVisible: false,
-                imageUri: image.uri,
-                imageName: image.name,
-                imageType: image.type,
+        const isValidRes = await isValidResolution(image);
+        if (!isValidRes) {
+            setError('avatarWithImagePicker.resolutionConstraints', {
+                minHeightInPx: CONST.AVATAR_MIN_HEIGHT_PX,
+                minWidthInPx: CONST.AVATAR_MIN_WIDTH_PX,
+                maxHeightInPx: CONST.AVATAR_MAX_HEIGHT_PX,
+                maxWidthInPx: CONST.AVATAR_MAX_WIDTH_PX,
             });
-        });
-    }
-
-    hideAvatarCropModal() {
-        this.setState({isAvatarCropModalOpen: false});
-    }
-
-    render() {
-        const DefaultAvatar = this.props.DefaultAvatar;
-        const additionalStyles = _.isArray(this.props.style) ? this.props.style : [this.props.style];
+            return;
+        }
 
-        return (
-            <View style={[styles.alignItemsCenter, ...additionalStyles]}>
-                <View style={[styles.pRelative, styles.avatarLarge]}>
+        setIsAvatarCropModalOpen(true);
+        setValidationError(null);
+        setPhraseParam({});
+        setIsMenuVisible(false);
+        setImageUri(image.uri);
+        setImageName(image.name);
+        setImageType(image.type);
```

Comment:
I'm seeing all of this are being used all together, I would create a single `useState` which uses an object, because those `setXXX` being inside of an async function React-Native can't batch the state updates, so right now will be 7 individual updates.

Reply:
U R right, except the `setIsMenuVisible` - updated